### PR TITLE
Feature/upgrade tf

### PR DESCRIPTION
**Pull request by stebenoist**
- Upgrade Terraform version to 0.11.8 for a4c_yorc_basic and a4c_yorc_ha topos.
- Create TerraformConfig node type to allow setting properties:
  1. plugins_dir
  2. plugins_download_urls
- Set default plugins_download_urls and plugins_dir for a4c_yorc_basic and a4c_yorc_ha topos.
- Install Terraform plugins in specified directory
- Config Yorc with Terraform plugins directory

### CODE REVIEW

File: org/ystia/yorc/yorc/pub/types.yaml [hunk 1]
```diff
@@ -169,6 +169,28 @@ node_types:
...
+    derived_from: tosca.nodes.Root
+    description: Terraform configuration for Yorc
+    abstract: true
+    properties:
+      plugins_dir:
+        description: "The directory where are located Terraform plugins"
+        type: string
+        required: true
+        default: /var/terraform/plugins
+      plugins_download_urls:
```
Comment (BertrandD) [inline org/ystia/yorc/yorc/pub/types.yaml]: I think we should add the downloads urls of plugins that are required by yorc as the default value of this property (as it is done in the topologies)

File: org/ystia/topologies/a4c_yorc_basic/types.yml [hunk 2]
```diff
@@ -217,7 +217,7 @@ topology_template:
     TerraformRuntime:
       type: org.ystia.terraform.linux.terraform.nodes.TerraformRuntime
       properties:
-        component_version: "0.9.11"
+        component_version: "0.11.8"
```
Comment (BertrandD) [inline org/ystia/topologies/a4c_yorc_basic/types.yml]: We should also update the default version of the TerraformRuntime component in org/ystia/terraform ?

File: org/ystia/yorc/yorc/pub/types.yaml [hunk 3]
```diff
@@ -169,6 +169,28 @@ node_types:
...
+        description: "List of Terraform plugins download urls"
+        type: list
+        required: true
+        entry_schema:
+          type: string
+    requirements:
+    - host:
+        capability: org.ystia.yorc.pub.capabilities.YorcConfigContainer
```
Comment (BertrandD) [inline org/ystia/yorc/yorc/pub/types.yaml]: Is it the good way of doing this ? Here you say that the TerraformConfig is hosted on Yorc. It's great, it works. But, I thought that it should be hosted on the TerraformRuntime as it is terraform that you are modifying.

The problem is, that if you change this to be hosted on TerraformRuntime, you cannot update the yorc config file to put the terraform plugins dir... We should discuss about it.

File: org/ystia/yorc/yorc/pub/types.yaml (hunk 1 above)
Comment (stebenoist) [reply org/ystia/yorc/yorc/pub/types.yaml]: ok

File: org/ystia/topologies/a4c_yorc_basic/types.yml (hunk 2 above)
Comment (stebenoist) [reply org/ystia/topologies/a4c_yorc_basic/types.yml]: Yes, I miss it

File: org/ystia/yorc/yorc/pub/types.yaml (hunk 3 above)
Comment (stebenoist) [reply org/ystia/yorc/yorc/pub/types.yaml]: Yes, furthermore, there is no straight relationship btw TerraformRuntime and the TF plugins.

**Comment by BertrandD** [comment]
LGTM

**Pull request merged**